server/src/main/java/io/crate/planner/operators/NestedLoopJoin.java

```diff
@@ -183,13 +189,9 @@ public ExecutionPlan build(DependencyCarrier executor,
         boolean expectedRowsAvailable = lhStats.numDocs() != -1 && rhStats.numDocs() != -1;
         if (expectedRowsAvailable) {
             if (!orderByWasPushedDown && joinType.supportsInversion() &&
-                (isDistributed && lhStats.numDocs() < rhStats.numDocs() && orderByFromLeft == null) ||
                 (blockNlPossible && lhStats.numDocs() > rhStats.numDocs())) {
```

Comment:
Why do we keep this one here, and not in the rule?

Reply:
`blockNlPossible` depends on properties from the execution plan(s) which are not available in the rule. I also prefer to do this step-by-step.